drift-protocol/src/main/java/com/facebook/drift/protocol/TProtocolUtil.java

```diff
@@ -94,4 +94,17 @@ public static void skip(TProtocolReader protocol, byte type)
                 throw new TProtocolException("Unknown type: " + type);
         }
     }
+
+    public static int readAllInBatches(TTransport transport, byte[] buf, int offset, int size)
+            throws TException
+    {
+        while (size > 0) {
+            // read data in 64KB chunks
+            int readSize = Math.min(size, 65536);
```

Comment:
instead of a magic number, I think we have can use new DataSize(64, KILOBYTE).toBytes;

Reply:
I think this constant is fine.

If you want to  - you can leave a couple of words about why this number is chosen, and why are we reading in chunks.